### PR TITLE
[JBPM-9073] Unique index error when correlationKey name is set to null

### PR DESCRIPTION
**Pull request by elguardian**
name in correlation key does not reflect the external form. Change to switch factory
implementation

### CODE REVIEW

**Comment by elguardian** [comment]
this should fix the problem. Please wait for CI to finish before retry the db matrix certification.

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java [hunk 1]
```diff
@@ -87,8 +90,7 @@ public void addProcessInstance(ProcessInstance processInstance, CorrelationKey c
         processInstanceInfo.updateLastReadDate();
         // generate correlation key if not given which is same as process instance id to keep uniqueness 
         if (correlationKey == null) {
-            correlationKey = new CorrelationKeyInfo();
-            ((CorrelationKeyInfo)correlationKey).addProperty(new CorrelationPropertyInfo(null, processInstanceInfo.getId().toString()));
+            correlationKey = (CorrelationKeyInfo) correlationKeyFactory.newCorrelationKey(processInstanceInfo.getId().toString());
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java]: it si equivalente code to https://github.com/kiegroup/jbpm/blob/master/jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/correlation/JPACorrelationKeyFactory.java#L32

**Comment by afalhambra** [comment]
jenkins execute downstream build

**Comment by elguardian** [comment]
Jenkins execute full downstream build

**Comment by elguardian** [comment]
failure is not related.
Jenkins execute full downstream build

**Comment by elguardian** [comment]
dear god...
Jenkins execute full downstream build

**Comment by elguardian** [comment]
@afalhambra unless you see something I don't. Those failures are not related. I would advise to merge this already.

**Comment by afalhambra** [comment]
> @afalhambra unless you see something I don't. Those failures are not related. I would advise to merge this already.

Yes, agree.

@kiegroup/gatekeepers please merge

**Pull request merged**